### DISCUSSION TITLE
Can someone please tell me or guide me to the documentation that explains how search results are ordered?

### DISCUSSION THREAD

**Discussion by steveh2023** [microsoft-search/pnp-modern-search#3827] [Q&A]
Hi,

We have a document library of a few thousand files and when doing a search for a certain keyword the 1st doc has this key word in 3 times and the 2nd has it 49 times.

So search order is not due to keyword frequency - so what exactly does determine the search order and how can it be configured - can it be configured?

**Comment by kasperbolarsen** [comment]
I assume you are using SharePoint as the data source? (SharePoint and Microsoft Search are not using the same ranking models)

You can tweak the results by using XRANK in both datasources

**Comment by wobba** [comment]
There is no documentation on how ranking works, nor will it be shared as it's a machine learned model. If you want a specific sort order, then sort on a property instead of the default relevance. There is some information on the dev docs at https://learn.microsoft.com/en-us/sharepoint/dev/general-development/sorting-search-results-in-sharepoint
